src/main/java/io/supertokens/emailpassword/EmailPassword.java

```diff
@@ -188,6 +188,27 @@ public static void resetPassword(Main main, String token, String password)
         }
     }
 
+    public static void updateUsersEmailOrPassword(Main main, @Nonnull String userId, @Nullable String email, @Nullable String password)
+            throws StorageQueryException, StorageTransactionLogicException {
+        EmailPasswordSQLStorage storage = StorageLayer.getEmailPasswordStorage(main);
+
+        storage.startTransaction(transaction -> {
+            if (email != null) {
```

Comment:
- You will first want to take a lock on all this userId's row by using a select for update query (which should already exist in the plugin interface).
- Then if the userId doesn't exist, you would return `"UNKNOWN_USER_ID_ERROR"`, else continue.